docs/template_plugin/tests/functional/op_reference/gru_sequence.cpp

```diff
@@ -0,0 +1,243 @@
+                                               params.hiddenSize,
+                                               params.gru_direction,
+                                               std::vector<std::string>{"sigmoid", "tanh"},
+                                               std::vector<float>{},
+                                               std::vector<float>{},
+                                               params.clip,
+                                               params.linear_before_reset);
+
+        auto function = std::make_shared<Model>(lstm_sequence->outputs(), ParameterVector{X, H_t, S_t, W, R, B});
+        return function;
+    }
+};
+
+TEST_P(ReferenceGRUSequenceTest, CompareWithRefs) {
+    Exec();
+}
+
+template <element::Type_t ET>
+std::vector<GRUSequenceParams> generateParamsRuntimeRef(size_t batchSize,
+                                                         size_t inputSize,
+                                                         size_t hiddenSize,
+                                                         size_t seqLength,
+                                                         float clip,
+                                                         bool linear_before_reset,
+                                                         op::RecurrentSequenceDirection gru_direction) {
+    using T = typename element_type_traits<ET>::value_type;
+
+    size_t numDirections = (gru_direction == op::RecurrentSequenceDirection::BIDIRECTIONAL) ? 2 : 1;
+
+    Shape XShape = Shape{batchSize, seqLength, inputSize};
+    Shape H_tShape = Shape{batchSize, numDirections, hiddenSize};
+    Shape S_tShape = Shape{batchSize};
+    Shape WShape = Shape{numDirections, 3 * hiddenSize, inputSize};
+    Shape RShape = Shape{numDirections, 3 * hiddenSize, hiddenSize};
+    Shape BShape;
+    Shape YShape = Shape{batchSize, numDirections, seqLength, hiddenSize};
+    Shape HoShape = Shape{batchSize, numDirections, hiddenSize};
+
+    if (linear_before_reset == true) {
+        BShape = Shape{numDirections, 4 * hiddenSize};
+    } else {
+        BShape = Shape{numDirections, 3 * hiddenSize};
+    }
+
+    std::vector<T> X = NGraphFunctions::Utils::generateVector<ET>(shape_size(XShape));
+    std::vector<T> H = NGraphFunctions::Utils::generateVector<ET>(shape_size(H_tShape));
+    std::vector<int64_t> S = NGraphFunctions::Utils::generateVector<element::Type_t::i64>(shape_size(S_tShape), seqLength);
+    std::vector<T> W = NGraphFunctions::Utils::generateVector<ET>(shape_size(WShape));
+    std::vector<T> R = NGraphFunctions::Utils::generateVector<ET>(shape_size(RShape));
+    std::vector<T> B = NGraphFunctions::Utils::generateVector<ET>(shape_size(BShape));
+
+    std::vector<std::string> activations = {"sigmoid", "tanh"};
+    std::vector<T> Y(shape_size(YShape));
+    std::vector<T> Ho(shape_size(HoShape));
+
+    ngraph::runtime::reference::gru_sequence<T, int64_t>(reinterpret_cast<const char*>(X.data()),
+                   XShape,
```

Comment:
@iefode @ilyachur I used runtime reference function of LSTMSequence to get refData because I cannot get reference output data from internal or external(ONNX page) source. Current implement is using random generated input as same as plugin functional test cases.. Now I can make dump of input/output from this function and replace to hard corded values. How do you think about this?

Reply:
Can we avoid the using of reference implementation for generation reference values? Because in this case the test checks nothing

Reply:
@ilyachur Got it! I will generate the input/refOut data from outside and put them as hard coded as other test cases.

Reply:
Used hard coded values for input/refOut data/